### PR TITLE
MOSIP-24463, MOSIP-24464, MOSIP-24465, MOSIP-24466   Updated the Mimoto scripts 

### PR DESCRIPTION
**Pull request by Sanjayss04**
Updated the Mimoto Helper and Test script.

### CODE REVIEW

File: mimoto/scripts/Mimoto_Helper_Script.jmx [hunk 1]
```diff
@@ -78,14 +66,19 @@
...
           <elementProp name="idRepoAppId" elementType="Argument">
             <stringProp name="Argument.name">idRepoAppId</stringProp>
             <stringProp name="Argument.value">idrepo</stringProp>
             <stringProp name="Argument.metadata">=</stringProp>
           </elementProp>
+          <elementProp name="addIdentitySetup" elementType="Argument">
+            <stringProp name="Argument.name">addIdentitySetup</stringProp>
+            <stringProp name="Argument.value">1</stringProp>
+            <stringProp name="Argument.metadata">=</stringProp>
+          </elementProp>
```
Comment (vishwa-vyom) [inline mimoto/scripts/Mimoto_Helper_Script.jmx]: what is this variable, name is confusing

File: mimoto/scripts/Mimoto_Helper_Script.jmx [hunk 2]
```diff
@@ -158,6 +151,16 @@
             <intProp name="Assertion.test_type">16</intProp>
           </ResponseAssertion>
           <hashTree/>
+          <ResponseAssertion guiclass="AssertionGui" testclass="ResponseAssertion" testname="Response Assertion - Response Code" enabled="true">
```
Comment (vishwa-vyom) [inline mimoto/scripts/Mimoto_Helper_Script.jmx:119]: I see we are creating a token with resident app id and using in all the mimoto APIs, do we need this ?

File: mimoto/support-files/uinVidList.txt [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+5627436418,6270374814657254
+7196340816,6046295391485721
+2186026301,4026720564731608
+8078526519,3758705206180148
+7625946193,5894852657207410
+9659170612,6396486219081031
+8367941524,2542612965135281
+2314851821,4803710935015965
+2079061970,5165817015302387
+2051478259,2406720417547968
```
Comment (vishwa-vyom) [inline mimoto/support-files/uinVidList.txt]: This will should get autogenerated as part of setup, why should we commit this ?

Comment (vishwa-vyom) [reply mimoto/support-files/uinVidList.txt]: this goes for other 2 txt files also

File: mimoto/scripts/Mimoto_Test_Script.jmx [hunk 4]
```diff
@@ -723,6 +761,952 @@ new File(&quot;TransactionID_RequestIndividual-ID.txt&quot;).delete();
...
+          <BeanShellPostProcessor guiclass="TestBeanGUI" testclass="BeanShellPostProcessor" testname="Store TransactionID To File" enabled="true">
+            <stringProp name="filename"></stringProp>
+            <stringProp name="parameters"></stringProp>
+            <boolProp name="resetInterpreter">false</boolProp>
+            <stringProp name="script">${__StringToFile(transactionId_CredentialRequest.txt,${UIN}\,${tranIdCr}\n,true,)}</stringProp>
+          </BeanShellPostProcessor>
+          <hashTree/>
+        </hashTree>
+      </hashTree>
+      <ThreadGroup guiclass="ThreadGroupGui" testclass="ThreadGroup" testname="Mimoto Request Credential API (Execution)" enabled="true">
```
Comment (vishwa-vyom) [inline mimoto/scripts/Mimoto_Test_Script.jmx]: Mimoto Request Credential API (Preparation) - Transaction ID Counter - CredentialRequest (UIN) - max value is same as others endpoints, will it not create issue when samples are high ?

File: mimoto/scripts/Mimoto_Test_Script.jmx [hunk 5]
```diff
@@ -723,6 +761,952 @@ new File(&quot;TransactionID_RequestIndividual-ID.txt&quot;).delete();
...
+        <stringProp name="ThreadGroup.on_sample_error">continue</stringProp>
+        <elementProp name="ThreadGroup.main_controller" elementType="LoopController" guiclass="LoopControlPanel" testclass="LoopController" testname="Loop Controller" enabled="true">
+          <boolProp name="LoopController.continue_forever">false</boolProp>
+          <stringProp name="LoopController.loops">1</stringProp>
```
Comment (vishwa-vyom) [inline mimoto/scripts/Mimoto_Test_Script.jmx:1431]: Mimoto Request Credential API (Preparation) - In thread group namings, i think we don't use the API at the back. Lets correct in all places

File: mimoto/scripts/Mimoto_Test_Script.jmx [hunk 6]
```diff
@@ -723,6 +761,952 @@ new File(&quot;TransactionID_RequestIndividual-ID.txt&quot;).delete();
...
+        </elementProp>
+        <stringProp name="ThreadGroup.num_threads">1</stringProp>
+        <stringProp name="ThreadGroup.ramp_time">1</stringProp>
```
Comment (vishwa-vyom) [inline mimoto/scripts/Mimoto_Test_Script.jmx:1434]: where are we using the vid generated in the setup process ?

File: mimoto/scripts/Mimoto_Helper_Script.jmx [hunk 7]
```diff
@@ -124,7 +117,7 @@
               </elementProp>
             </collectionProp>
           </elementProp>
```
Comment (vishwa-vyom) [inline mimoto/scripts/Mimoto_Helper_Script.jmx:84]: In eSignet, I believe we moved the setup process to the same test script file, can we do the same for mimoto also ?

File: mimoto/scripts/Mimoto_Helper_Script.jmx [hunk 8]
```diff
@@ -124,7 +117,7 @@
...
-          <stringProp name="HTTPSampler.domain">${serverIP}</stringProp>
+          <stringProp name="HTTPSampler.domain">${serverIPInternal}</stringProp>
           <stringProp name="HTTPSampler.port">${serverPortNo}</stringProp>
           <stringProp name="HTTPSampler.protocol">${protocol}</stringProp>
```
Comment (vishwa-vyom) [inline mimoto/scripts/Mimoto_Helper_Script.jmx:87]: Mimoto Req IndividualID OTP (Preparation) & Mimoto Aid Get-Individual-ID (Execution) are these related? naming is different here

File: mimoto/scripts/Mimoto_Helper_Script.jmx [hunk 9]
```diff
@@ -38,17 +26,17 @@
           </elementProp>
           <elementProp name="residentSecretKey" elementType="Argument">
             <stringProp name="Argument.name">residentSecretKey</stringProp>
-            <stringProp name="Argument.value">abc123</stringProp>
```
Comment (vishwa-vyom) [inline mimoto/scripts/Mimoto_Helper_Script.jmx:41]: Readme file to be added in the same format as eSignet

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: mimoto/scripts/Mimoto_Helper_Script.jmx (hunk 1 above)
Comment (Sanjayss04) [reply mimoto/scripts/Mimoto_Helper_Script.jmx]: We have been using this variable name only where ever we are using the identity creation setup.

File: mimoto/support-files/uinVidList.txt (hunk 3 above)
Comment (Sanjayss04) [reply mimoto/support-files/uinVidList.txt]: Okay @vishwa-vyom Removed both files.

File: mimoto/scripts/Mimoto_Helper_Script.jmx (hunk 7 above)
Comment (Sanjayss04) [reply mimoto/scripts/Mimoto_Helper_Script.jmx:84]: Moved the setup process to test script.

File: mimoto/scripts/Mimoto_Helper_Script.jmx (hunk 8 above)
Comment (Sanjayss04) [reply mimoto/scripts/Mimoto_Helper_Script.jmx:87]: Yes both are related, Mimoto Req IndividualID OTP api is required for the praparation so have added one more separate thread group for Mimoto Req IndividualID OTP (Execution).

File: mimoto/scripts/Mimoto_Test_Script.jmx (hunk 5 above)
Comment (Sanjayss04) [reply mimoto/scripts/Mimoto_Test_Script.jmx:1431]: Updated

File: mimoto/scripts/Mimoto_Test_Script.jmx (hunk 4 above)
Comment (Sanjayss04) [reply mimoto/scripts/Mimoto_Test_Script.jmx]: I think we are using this same pattern everywhere, and the difference between all the Transaction ID Counter is 100M. I believe that should be fine.

File: mimoto/scripts/Mimoto_Test_Script.jmx (hunk 6 above)
Comment (Sanjayss04) [reply mimoto/scripts/Mimoto_Test_Script.jmx:1434]: Currently we are not using the generated VID anywhere. Will remove it from the setup process.

File: mimoto/scripts/Mimoto_Helper_Script.jmx (hunk 2 above)
Comment (Sanjayss04) [reply mimoto/scripts/Mimoto_Helper_Script.jmx:119]: The auth token generated with resident app id was used for authentication purpose and from starting we have added that in the header for all the api's. This thing I have to once check it by removing it. Will update once done.

Comment (vishwa-vyom) [reply mimoto/scripts/Mimoto_Helper_Script.jmx:119]: Do we have the update on this ?

**Review by vishwa-vyom — APPROVED** [review]

**Pull request merged**